### ISSUE TITLE
git required for building from source tarball

### ISSUE THREAD

**Issue by afaerber** [ARM-software/ComputeLibrary#17]
If the `git` executable is not installed on the build host, `scons` errors out:

```
...
[   49s]   File "/usr/lib/python2.7/site-packages/SCons/Script/SConscript.py", line 604:
[   49s]     return method(*args, **kw)
[   49s]   File "/usr/lib/python2.7/site-packages/SCons/Script/SConscript.py", line 541:
[   49s]     return _SConscript(self.fs, *files, **subst_kw)
[   49s]   File "/usr/lib/python2.7/site-packages/SCons/Script/SConscript.py", line 250:
[   49s]     exec _file_ in call_stack[-1].globals
[   49s]   File "/home/abuild/rpmbuild/BUILD/ComputeLibrary-17.03.1/sconscript", line 65:
[   49s]     git_hash = subprocess.check_output(["git", "rev-parse","HEAD"])
[   49s]   File "/usr/lib/python2.7/subprocess.py", line 212:
[   49s]     process = Popen(stdout=PIPE, *popenargs, **kwargs)
[   49s]   File "/usr/lib/python2.7/subprocess.py", line 390:
[   49s]     errread, errwrite)
[   49s]   File "/usr/lib/python2.7/subprocess.py", line 1024:
[   49s]     raise child_exception
```

That should not be necessary since this is what its presence results in:

```
[  164s] scons: Reading SConscript files ...
[  165s] fatal: Not a git repository (or any of the parent directories): .git
[  167s] scons: done reading SConscript files.
[  167s] scons: Building targets ...
```

I.e., `git` does not get any information from the tarball, so it is rather useless for release builds. Requiring to install unneeded packages is an inconvenience in an OBS build environment, as it adds to build time for each package rebuild.

Probably this could be prevented by adding some `except` clause below?

```
git_hash="unknown"
try:
    git_hash = subprocess.check_output(["git", "rev-parse","HEAD"])
except subprocess.CalledProcessError:
    pass
```

**Comment by AnthonyARM** [comment]
Yes, we've noticed this issue too, will be fixed in the next maintenance release.
Thanks for reporting it

**Comment by AnthonyARM** [comment]
Fixed in v17.04 (Re-open if you still see the issue)
